src/app/GitCommands/UserRepositoryHistory/RecentRepoInfo.cs

```diff
@@ -118,7 +125,7 @@ void AddSortedRepos(bool mostRecent, List<RecentRepoInfo> addToList)
                 addToList.AddRange(
                     from caption in orderedRepos.Keys
                     from repo in orderedRepos[caption]
-                    where !mostRecent || repo.MostRecent == mostRecent
+                    where (!mostRecent && !HideTopRepositoriesFromRecentList) || repo.MostRecent == mostRecent
```

Comment:
`mostRecent` and `MostRecent` should be renamed to `TopRepo`.
```suggestion
                    where repo.TopRepo == topRepo || (!topRepo && !HideTopRepositoriesFromRecentList) 
```